process/transaction/shardProcess.go

```diff
@@ -1293,16 +1307,23 @@ func (txProc *txProcessor) createCompleteEventLog(scr data.TransactionHandler, o
 	}
 }
 
-func (txProc *txProcessor) saveFailedLogsIfNeeded(originalTxHash []byte) {
-	logsTx, logs, ok := txProc.failedTxLogsAccumulator.GetLogs(originalTxHash)
-	if ok {
-		ignorableErr := txProc.txLogsProcessor.SaveLog(originalTxHash, logsTx, logs)
-		if ignorableErr != nil {
-			log.Debug("txLogsProcessor.SaveLog failed", "error", ignorableErr.Error())
-		}
+func (txProc *txProcessor) createCompletedRelayedTxV3EventLog(tx data.TransactionHandler, originalTxHash []byte) {
+	if len(txProc.relayedTxV3InnerHashes) == 0 {
```

Comment:
can you change this and accumulate the inner hashes into a structure which is not a local variable for the txProc ?

I think it can be done.